### PR TITLE
WSTEAMA-586: Most Read component on Index (CPS IDX) Pages

### PR DESCRIPTION
**Pull request by holchris**
Resolves JIRA https://jira.dev.bbc.co.uk/browse/WSTEAMA-586

Overall changes
======
Implement the new Most Read component on Index Pages
Fetch IDX page data from the BFF (instead of json endpoint)

Code changes
======
- Swap legacy most read component for new component in IdxPage
- Update idx/getInitialData to fetch from .json on local environment, but BFF on test/live environment
- Update fixture data / tests / stories accordingly 

Testing
======
- [ ] http://localhost:7080/persian/afghanistan displays a most read component
- [ ] http://localhost:7080/ukrainian/ukraine_in_russian does not display a most read component (because the ukrainian service has the mostReadOnIdxPage toggle set to false) 

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: src/app/pages/IdxPage/IdxPage.jsx [hunk 1]
```diff
@@ -88,6 +64,19 @@ const IdxPage = ({
...
   const radioScheduleIdxPosition = path(['idxPagePosition'], radioSchedule);
 
+  const { mostRead: mostReadInitialData } = pageData;
+
+  const renderMostRead = () => (
+    <IdxMostReadSection>
+      <MostRead
+        data={mostReadInitialData}
+        columnLayout="twoColumn"
+        size="default"
```
Comment (karinathomasbbc) [inline src/app/pages/IdxPage/IdxPage.jsx]: This is already the default option, so we don't need to explicitly add this

```suggestion
```

File: src/app/pages/IdxPage/IdxPage.jsx [hunk 2]
```diff
@@ -53,31 +51,9 @@ const StyledRadioScheduleContainer = styled(RadioScheduleContainer)`
...
-};
-
-const IdxPage = ({
-  pageData,
-  mostReadEndpointOverride,
-  radioScheduleEndpointOverride,
-}) => {
+const IdxPage = ({ pageData, radioScheduleEndpointOverride }) => {
   const { mostRead, lang, radioSchedule } = useContext(ServiceContext);
+  console.log({ pageData });
```
Comment (karinathomasbbc) [inline src/app/pages/IdxPage/IdxPage.jsx]:
```suggestion
```

File: src/app/routes/idx/getInitialData/index.js [hunk 3]
```diff
@@ -36,25 +43,52 @@ export const hasRadioSchedule = async (service, variant) => {
...
+      pathname,
+      pageType: CPS_ASSET,
+      service,
+      variant,
+    });
+
+    const optHeaders = { 'ctx-service-env': getEnvironment(pathname) };
     const pageHasRadioSchedule = await hasRadioSchedule(service, variant);
-    const pageDataPromise = fetchPageData({ path, pageType });
+    console.log({ pageHasRadioSchedule });
```
Comment (karinathomasbbc) [inline src/app/routes/idx/getInitialData/index.js]:
```suggestion
```

File: src/app/routes/idx/getInitialData/index.js [hunk 4]
```diff
@@ -36,25 +43,52 @@ export const hasRadioSchedule = async (service, variant) => {
...
       status,
-      pageData: transformJson(json),
+      pageData: {
+        ...transformJson(json?.data?.article),
+        radioScheduleData: json?.radioScheduleData,
+        mostRead: json?.data?.secondaryData?.mostRead,
+      },
     };
   } catch ({ message, status = getErrorStatusCode() }) {
+    console.error(message);
```
Comment (karinathomasbbc) [inline src/app/routes/idx/getInitialData/index.js]:
```suggestion
```

File: src/integration/pages/mostWatchedPage/afrique/__snapshots__/amp.test.js.snap
Comment (karinathomasbbc) [inline src/integration/pages/mostWatchedPage/afrique/__snapshots__/amp.test.js.snap:1]: Not sure why these have changed, so perhaps need to revert this file (and others which are not `idxPage` related?

**Review by amoore108 — APPROVED** [review]
Nice!

**Review by karinathomasbbc — APPROVED** [review]

**Pull request merged**